### ISSUE TITLE
set default hypothesis settings for `deadline` to `None`

### ISSUE THREAD

**Issue by bout3fiddy** [vyperlang/titanoboa#37]
Currently, one writes fuzzing tests in the following manner:

```
from hypothesis import settings, given
from boa.test import strategy


@given(val=strategy("uint256[3]"))
@settings(max_examples=10, deadline=None)
def test_something(some_fixture, val):
    assert val is not None
```

If one does not provide `None` as `deadline`, then tests will bork because of flakiness. This flakiness can arise from inconsistent test times when hypothesis does runs tests with the same inputs.

To prevent this, perhaps boa should have a default deadline time set to `None`.

**Comment by charles-cooper** [comment]
hmm. on the one hand i am not really opposed to this. on the other hand -- people who are used to hypothesis will probably already be accustomed to configuring this in their project, or on a per-test basis, and might end up confused as to why we are changing the default settings

**Comment by DanielSchiavini** [comment]
Why would the test time be so different? If we are running the same code it should take just as long. Is something being cached?

**Comment by DanielSchiavini** [comment]
that is a general hypothesis issue then, totally not specific to boa?

**Comment by charles-cooper** [comment]
> that is a general hypothesis issue then, totally not specific to boa?

right. it comes down to whether people think of boa and hypothesis as "partner" packages, or whether boa is more of a wrapper around hypothesis (like boa chooses the fuzzer backend; hypothesis is an implementation detail)
